### PR TITLE
DevOps roles not been initialed

### PR DESCRIPTION
**Pull request by RolandMa1986**
Signed-off-by: Roland.Ma <rolandma@yunify.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Initial Roles for namespace is a KubeSphere Project or DevOps Project
**Which issue(s) this PR fixes**:
#3395
**Special notes for reviewers**:
ref https://github.com/kubesphere/kubesphere/pull/3179

### CODE REVIEW

File: pkg/controller/namespace/namespace_controller.go [hunk 1]
```diff
@@ -129,19 +130,22 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
-		if err := r.initRoles(rootCtx, logger, namespace); err != nil {
-			return ctrl.Result{}, err
-		}
 	} else {
 		if err := r.unbindWorkspace(rootCtx, logger, namespace); err != nil {
 			return ctrl.Result{}, err
 		}
 	}
+	// Need to init roles for devops/project namespaces
+	if _, ok := namespace.Labels[constants.NamespaceLabelKey]; ok {
```
Comment (wansir) [inline pkg/controller/namespace/namespace_controller.go:144]: `NamespaceLabelKey` will be present in all namespaces. We need to distinguish between namesapce created through ks-console and namespace created directly through kubectl, which can be judged by `WorkspaceLabelKey` and `DevOpsProjectLabelKey`or create builtin roles for all namespaces.

Comment (RolandMa1986) [reply pkg/controller/namespace/namespace_controller.go:144]: @wansir, maybe we can change to following
```
	_, hasWorkspaceLabel := namespace.Labels[tenantv1alpha1.WorkspaceLabel]
	_, hasDevOpsProjectLabel := namespace.Labels[constants.DevOpsProjectLabelKey]
	if hasWorkspaceLabel || hasDevOpsProjectLabel {
```

Comment (wansir) [reply pkg/controller/namespace/namespace_controller.go:144]: Agreed, this should be part of initRoles.

**Comment by zryfish** [comment]
Any more comments?

**Pull request merged**